app/models/theme.rb

```diff
@@ -241,6 +241,19 @@ def self.user_theme_ids
     end
   end
 
+  def self.enabled_theme_and_component_ids
+    get_set_cache "enabled_theme_and_component_ids" do
+      theme_ids = Theme.user_selectable.where(enabled: true).pluck(:id)
+      component_ids =
+        ChildTheme
+          .where(parent_theme_id: theme_ids)
+          .joins(:child_theme)
+          .where(themes: { enabled: true })
+          .pluck(:child_theme_id)
+      (theme_ids + component_ids).uniq
```

Comment:
You can use the `|` operator to do `+` and `.uniq` 😉 

```suggestion
      theme_ids | component_ids
```

Reply:
Ah, nice bit of code golf ⛳, that! Thanks!